drivers/spi/spi_b_renesas_ra8.c

```
@@ -0,0 +1,771 @@
+
+static int ra_spi_b_configure(const struct device *dev, const struct spi_config *config)
+{
+	struct ra_spi_data *data = dev->data;
+	fsp_err_t fsp_err;
+
+	if (spi_context_configured(&data->ctx, config)) {
+		/* Nothing to do */
+		return 0;
+	}
+
+	fsp_err = R_SPI_B_Close(&data->spi);
+
+	if ((config->operation & SPI_FRAME_FORMAT_TI) == SPI_FRAME_FORMAT_TI) {
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_OP_MODE_SLAVE) {
+		data->fsp_config.operating_mode = SPI_MODE_SLAVE;
+	} else {
+		data->fsp_config.operating_mode = SPI_MODE_MASTER;
+	}
+
+	if (SPI_MODE_GET(config->operation) & SPI_MODE_CPOL) {
+		data->fsp_config.clk_polarity = SPI_CLK_POLARITY_HIGH;
+	} else {
+		data->fsp_config.clk_polarity = SPI_CLK_POLARITY_LOW;
+	}
+
+	if (SPI_MODE_GET(config->operation) & SPI_MODE_CPHA) {
+		data->fsp_config.clk_phase = SPI_CLK_PHASE_EDGE_EVEN;
+	} else {
+		data->fsp_config.clk_phase = SPI_CLK_PHASE_EDGE_ODD;
+	}
+
+	if (config->operation & SPI_TRANSFER_LSB) {
+		data->fsp_config.bit_order = SPI_BIT_ORDER_LSB_FIRST;
+	} else {
+		data->fsp_config.bit_order = SPI_BIT_ORDER_MSB_FIRST;
+	}
+
+	if (config->frequency > 0) {
+		fsp_err = R_SPI_B_CalculateBitrate(config->frequency,
+						   data->fsp_config_extend.clock_source,
+						   &data->fsp_config_extend.spck_div);
+		__ASSERT(fsp_err == 0, "spi_b: spi frequency calculate error: %d", fsp_err);
+	}
+
+	data->fsp_config_extend.spi_comm = SPI_B_COMMUNICATION_FULL_DUPLEX;
+	if (spi_cs_is_gpio(config) || !IS_ENABLED(CONFIG_SPI_B_USE_HW_SS)) {
+		data->fsp_config_extend.spi_clksyn = SPI_B_SSL_MODE_CLK_SYN;
+	} else {
+		data->fsp_config_extend.spi_clksyn = SPI_B_SSL_MODE_SPI;
+		data->fsp_config_extend.ssl_select = SPI_B_SSL_SELECT_SSL0;
+	}
+
+	data->fsp_config.p_extend = &data->fsp_config_extend;
+
+	data->fsp_config.p_callback = spi_cb;
+	data->fsp_config.p_context = dev;
+	fsp_err = R_SPI_B_Open(&data->spi, &data->fsp_config);
+	if (fsp_err != FSP_SUCCESS) {
+		LOG_ERR("R_SPI_B_Open error: %d", fsp_err);
```

Should we return an error?